src/genpy/rostime.py

```diff
@@ -395,6 +396,11 @@ def __floordiv__(self, val):
         """
         t = type(val)
         if t in (int, long):
+            warnings.warn(
+                'The Duration.__floordiv__(integer) function is ill-defined. '
+                'The floor operation is applied independently on the seconds as well as the nanoseconds. '
+                'For true division use a float divisor or the operator `/` (which requires `from __future__ import division` in Python 2) or `__truediv__` instead.',
```

Comment:
I have removed the comment about the float divisor from `__floordiv__`. That only makes sense for `__div__` where using floats actually makes a difference.